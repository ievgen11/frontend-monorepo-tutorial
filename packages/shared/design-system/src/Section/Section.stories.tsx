import * as React from "react";
import { withKnobs } from "@storybook/addon-knobs";

import image from "./image.png";
import { Section } from "./index";

export default {
  title: "Section",
  component: Section,
  decorators: [withKnobs],
};

export const Default = () => (
  <Section image={image} >
    <h1>This is an example section</h1>
    <p>Notice the nice image on the right!</p>
  </Section>
);
