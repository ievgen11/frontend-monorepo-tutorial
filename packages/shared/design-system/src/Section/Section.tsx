import React from "react";
import styled from "styled-components";

const Wrapper = styled.section`
  width: 100%;
  min-height: 75vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Container = styled.div`
  width: 100%;
  padding: 20px 24px 30px 24px;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;

  @media only screen and (min-width: 768px) {
    width: 100%;
    flex-direction: row;
    margin: auto;
    padding: 64px 80px 30px 80px;
    max-width: 900px;
  }
`;

const Content = styled.div`
  flex-grow: 1;
  width: 100%;
`;

const Image = styled.img`
  max-width: 150px;
  margin: 30px 0 0 0;

  @media only screen and (min-width: 600px) {
    max-width: 250px;
    margin: 0 0 0 60px;
  }
`;

export interface SectionProps {
  children?: React.ReactNode;
  image?: string;
}

export const Section: React.FC<SectionProps> = ({
  children,
  image,
}): JSX.Element => (
  <Wrapper>
    <Container>
      <Content>{children}</Content>
      {image ? <Image src={image} /> : null}
    </Container>
  </Wrapper>
);
