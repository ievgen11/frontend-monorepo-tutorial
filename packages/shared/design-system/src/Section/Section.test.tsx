import * as React from "react";
import * as renderer from "react-test-renderer";

import { Section } from "./Section";

describe("<Section />", () => {
  test("Snapshot is intact", () => {
    const tree = renderer.create(<Section>Test</Section>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
