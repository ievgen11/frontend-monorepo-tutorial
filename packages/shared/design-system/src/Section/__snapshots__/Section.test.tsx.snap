// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> Snapshot is intact 1`] = `
<section
  className="sc-bdfBwQ fpQbfy"
>
  <div
    className="sc-gsTCUz fcuqMu"
  >
    <div
      className="sc-dlfnbm bcBhUh"
    >
      Test
    </div>
  </div>
</section>
`;
